src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts

```diff
@@ -567,17 +614,46 @@ export function parseGeneticInput(input:string):{status:"complete", result:Oncop
                             throw new Error(`${errorPrefix}Alteration "${alteration}" is not valid - it must be "HIGH" or "LOW" if Type is "PROT"`);
                         }
                         break;
+                    case "fusion":
+                        if (lcType !== "fusion") {
+                            throw new Error(`${errorPrefix}Type "${type}" is not valid - it must be "FUSION" if Alteration is "FUSION"`);
+                        } else {
+                            ret.alteration = lcType as OncoprintMutationType;
+                            ret.proteinChange = alteration;
+                        }
+                        break;
                     default:
                         // everything else is a mutation
-                        if (["missense", "inframe", "fusion", "promoter", "trunc", "other"].indexOf(lcType) === -1) {
-                            if (lcAlteration === "fusion") {
-                                throw new Error(`${errorPrefix}Type "${type}" is not valid - it must be "FUSION" if Alteration is "FUSION"`);
-                            } else {
-                                throw new Error(`${errorPrefix}Type "${type}" is not valid - it must be "MISSENSE", "INFRAME", "TRUNC", "PROMOTER", or "OTHER" for a mutation alteration.`);
+                        // use OQL parsing for handling mutation modifiers
+                        let parsedMutation:MUTCommand<any>;
+                        try {
+                            parsedMutation = (parseOQLQuery(`GENE: ${lcType}`)[0].alterations as Alteration[])[0] as MUTCommand<any>;
+                        } catch (e) {
+                            throw new Error(`${errorPrefix}Mutation type ${type} is not valid.`);
+                        }
+
+                        for (const modifier of parsedMutation.modifiers) {
+                            switch (modifier) {
+                                case "GERMLINE":
+                                    ret.isGermline = true;
+                                    break;
+                                case "DRIVER":
+                                    ret.isCustomDriver = true;
+                                    break;
+                                default:
+                                    throw new Error(`${errorPrefix}Only allowed mutation modifiers are GERMLINE and DRIVER`);
```

Comment:
it's nice that you are throwing these errors.  we should discuss how we are presenting these kind of errors to user.  i know we discussed error boundaries before.  i think you found that they are not compatible with mobx.  lets discuss

Reply:
I don't see the need for an error boundary here, but we can discuss